src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs

```diff
@@ -206,14 +212,15 @@ public virtual async Task<T> GetConfigurationAsync(CancellationToken cancel)
                 }
                 finally
                 {
+                    _configurationRetrieverState = ConfigurationRetrieverIdle;
                     _configurationNullLock.Release();
                 }
-#pragma warning restore CA1031 // Do not catch general exception types
             }
             else
             {
                 if (Interlocked.CompareExchange(ref _configurationRetrieverState, ConfigurationRetrieverRunning, ConfigurationRetrieverIdle) == ConfigurationRetrieverIdle)
                 {
+                    NumberOfTimesAutomaticRefreshRequested++;
                     _ = Task.Run(UpdateCurrentConfiguration, CancellationToken.None);
```

Comment:
Can we please rename `UpdateCurrentConfiguration` to `RetrieveAndUpdateConfiguration`? So that it's more distinct from `UpdateConfiguration` which updates the config instance variable.